File: tarea4/vaciarMatriz.py
#!/usr/bin/env python3
import typer

app = typer.Typer(invoke_without_command= True)

matriz = [
    [1,2,3],
    [4,5,6],
    [7,8,9],
]

arreglo = []

@app.callback()
def main(ctx: typer.Context):
    if ctx.invoked_subcommand is None:
        print(matriz)

@app.command()
def filas():
    for fila in matriz:
        for elemento in fila:
            arreglo.append(elemento)
    print(arreglo)

@app.command()
def columnas():
    for i in range(len(matriz)):
        for j in range(len(matriz[i])):
            arreglo.append(matriz[j][i])
    print(arreglo)

if __name__ == "__main__":
    app()

File: tarea4/vaciarMatriz.js
#!/usr/bin/env node

let matriz = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
];

let arreglo = [];
let args = process.argv.slice(2);

if (args.length === 0) {
    console.log(matriz);
} else {
    switch (args[0]) {
        case "filas": {
            matriz.forEach(fila => fila.forEach(e => arreglo.push(e)));
            console.log(arreglo);
            break;
        }
        case "columnas": {
            for (let i = 0; i < matriz[0].length; i++) {
                for (let j = 0; j < matriz.length; j++) {
                    arreglo.push(matriz[j][i]);
                }
            }
            console.log(arreglo);
            break;
        }
    }
}
